File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if (head == NULL || head->next == NULL) {
            return head;
        }
        ListNode* headnext = head->next;
        ListNode* headnextnext = head->next->next;
        headnext->next = NULL;
        ListNode* transformed = swapPairs(headnextnext);
        head->next = transformed;
        headnext->next = head;
        return headnext;

    }
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    if (head==null || head.next==null){
        return head
    }
    var temp = head.next.next;
    var next = head.next;
    next.next = head;
    head.next = swapPairs(temp);
    return next
};